File: check_database.js
const { createClient } = require('@supabase/supabase-js');
require('dotenv').config();

const supabase = createClient(process.env.SUPABASE_URL, process.env.SUPABASE_SERVICE_KEY);

async function checkDatabase() {
  try {
    console.log('🔍 Checking your Supabase database...\n');
    
    // Try to get brands
    console.log('Checking core.brands...');
    const { data: brands, error: brandsError } = await supabase
      .from('core.brands')
      .select('brand_name, brand_slug')
      .limit(3);
    
    if (brandsError) {
      console.log('❌ core.brands error:', brandsError.message);
    } else {
      console.log('✅ core.brands found:', brands.length, 'brands');
      if (brands.length > 0) {
        brands.forEach(brand => console.log('  -', brand.brand_name, '(' + brand.brand_slug + ')'));
      }
    }
    
    // Try to get whatsapp messages
    console.log('\nChecking signals.whatsapp_messages...');
    const { data: whatsapp, error: whatsappError } = await supabase
      .from('signals.whatsapp_messages')
      .select('signal_id, sender_phone, message_text')
      .limit(3);
    
    if (whatsappError) {
      console.log('❌ signals.whatsapp_messages error:', whatsappError.message);
    } else {
      console.log('✅ signals.whatsapp_messages found:', whatsapp.length, 'messages');
      if (whatsapp.length > 0) {
        whatsapp.forEach(msg => console.log('  -', msg.sender_phone, ':', msg.message_text.substring(0, 50) + '...'));
      }
    }
    
    // Try to get actors
    console.log('\nChecking actors.actors...');
    const { data: actors, error: actorsError } = await supabase
      .from('actors.actors')
      .select('actor_id, primary_identifier')
      .limit(3);
    
    if (actorsError) {
      console.log('❌ actors.actors error:', actorsError.message);
    } else {
      console.log('✅ actors.actors found:', actors.length, 'actors');
      if (actors.length > 0) {
        actors.forEach(actor => console.log('  -', actor.primary_identifier));
      }
    }
    
  } catch (error) {
    console.log('❌ Error:', error.message);
  }
}

checkDatabase();

File: intelligence_layer/check_database.py
#!/usr/bin/env python3
"""
Database Inspector - Check what signals are available
"""

import os
import sys
from dotenv import load_dotenv
from supabase import create_client, Client

# Load environment variables
load_dotenv()

def check_database():
    """Check database connection and available signals"""
    print("🔍 Database Inspector")
    print("=" * 50)
    
    # Get credentials
    url = os.getenv('SUPABASE_URL')
    key = os.getenv('SUPABASE_KEY')
    
    if not url or not key:
        print("❌ Missing SUPABASE_URL or SUPABASE_KEY")
        return
    
    print(f"✅ URL: {url}")
    print(f"✅ KEY: {key[:20]}...")
    
    try:
        # Create client
        client = create_client(url, key)
        print("✅ Supabase client created")
        
        # Check drivers table
        print("\n📊 Checking drivers table...")
        drivers = client.table('drivers').select('driver_name').execute()
        print(f"✅ Found {len(drivers.data)} drivers: {[d['driver_name'] for d in drivers.data]}")
        
        # Prefer unified view if present
        print("\n📊 Checking unified signals view...")
        try:
            unified = client.table('signals_unified').select('signal_id, signal_text, signal_type, source_platform').limit(5).execute()
            if unified.data:
                print(f"✅ signals_unified: {len(unified.data)} sample rows")
                for row in unified.data:
                    preview = (row.get('signal_text') or '')[:50]
                    print(f"   {row.get('signal_id')} [{row.get('signal_type')}/{row.get('source_platform')}]: {preview}...")
            else:
                print("❌ signals_unified: No rows (view empty)")
        except Exception as e:
            print(f"❌ signals_unified: Error - {str(e)}")

        print("\n📊 Checking raw signal tables...")
        raw_checks = [
            ('whatsapp_messages', 'message_text'),
            ('reviews', 'review_text'),
            ('signals', 'raw_content')
        ]
        for table, col in raw_checks:
            try:
                result = client.table(table).select(f'signal_id, {col}').limit(5).execute()
                if result.data:
                    print(f"✅ {table}: {len(result.data)} signals found")
                    first = result.data[0]
                    content = first.get(col) or ''
                    print(f"   Example: {first.get('signal_id')} - {str(content)[:50]}...")
                else:
                    print(f"❌ {table}: No signals found")
            except Exception as e:
                print(f"❌ {table}: Error - {str(e)}")
        
        # Check actor tables
        print("\n📊 Checking actor tables...")
        try:
            actors = client.table('actor_profiles').select('actor_id').limit(5).execute()
            print(f"✅ actor_profiles: {len(actors.data)} actors found")
        except Exception as e:
            print(f"❌ actor_profiles: Error - {str(e)}")
        
        print("\n🎯 Ready to process signals!")
        
    except Exception as e:
        print(f"❌ Database error: {str(e)}")

if __name__ == "__main__":
    check_database()
